File: src/main/java/ml/duncte123/skybot/commands/fun/DialogCommand.java
/*
 *
 * MIT License
 *
 * Copyright (c) 2017 Duncan Sterken
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package ml.duncte123.skybot.commands.fun;

import ml.duncte123.skybot.objects.command.Command;
import ml.duncte123.skybot.utils.EmbedUtils;
import ml.duncte123.skybot.utils.Settings;
import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
import org.apache.commons.lang3.StringUtils;
import org.apache.commons.text.WordUtils;

import java.util.Arrays;

public class DialogCommand extends Command {

    /**
     * {@inheritDoc}
     */
    @Override
    public void executeCommand(String[] args, GuildMessageReceivedEvent event) {

        if (args.length < 1) {
            event.getChannel().sendMessage("Correct usage: `"+ Settings.prefix+getName()+" <words>`").queue();
            return;
        }

        String linesBeforeWrap = StringUtils.join(args, " ").replaceAll("`", "");

        String lines = WordUtils.wrap(linesBeforeWrap, 25, null, true);

        String[] split = lines.split("\n");

        StringBuilder sb = new StringBuilder()
                .append("```")
                .append("╔═══════════════════════════╗ \n")
                .append("║ Alert                     ║\n")
                .append("╠═══════════════════════════╣\n");
                //.append("║ " + String.format("%-25s", lines) + " ║\n") //Thnx Kantenkugel#1568 you're awesome
                Arrays.stream(split).map(String::trim).map( it -> String.format("%-25s", it) ).map(it -> "║ "+it+" ║\n").forEach(sb::append);
              sb.append("║  ┌─────────┐  ┌────────┐  ║\n")
                .append("║  │   Yes   │  │   No   │  ║\n")
                .append("║  └─────────┘  └────────┘  ║\n")
                .append("╚═══════════════════════════╝\n")
                .append("```");
        sendEmbed(event, EmbedUtils.embedMessage(sb.toString()));

    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String help() {
        return "Gives you a nice dialog";
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String getName() {
        return "dialog";
    }
}

File: src/main/kotlin/ml/duncte123/skybot/commands/fun/DialogCommand.kt
/*
 * Skybot, a multipurpose discord bot
 *      Copyright (C) 2017 - 2020  Duncan "duncte123" Sterken & Ramid "ramidzkh" Khan & Maurice R S "Sanduhr32"
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package ml.duncte123.skybot.commands.`fun`

import me.duncte123.botcommons.messaging.MessageUtils
import ml.duncte123.skybot.Author
import ml.duncte123.skybot.objects.command.Command
import ml.duncte123.skybot.objects.command.CommandCategory
import ml.duncte123.skybot.objects.command.CommandContext
import org.apache.commons.text.WordUtils

@Author(nickname = "Sanduhr32", author = "Maurice R S")
class DialogCommand : Command() {

    init {
        this.category = CommandCategory.FUN
        this.name = "dialog"
        this.help = "Displays a confirmation dialog"
        this.usage = "<text>"
    }

    override fun execute(ctx: CommandContext) {
        if (ctx.args.isEmpty()) {
            this.sendUsageInstructions(ctx)
            return
        }

        val lines = WordUtils.wrap(
            ctx.argsDisplay.replace("`", ""),
            25,
            null,
            true
        ).split("\n")

        val string = buildString {
            appendLine("```")
            appendLine("╔═══════════════════════════╗ ")
            appendLine("║ Alert                     ║")
            appendLine("╠═══════════════════════════╣")

            lines.stream()
                .map { it.trim() }
                .map { String.format("%-25s", it) }
                .map { "║ $it ║" }
                .forEach { appendLine(it) }

            appendLine("║  ┌─────────┐  ┌────────┐  ║")
            appendLine("║  │   Yes   │  │   No   │  ║")
            appendLine("║  └─────────┘  └────────┘  ║")
            appendLine("╚═══════════════════════════╝")
            appendLine("```")
        }

        MessageUtils.sendMsg(ctx, string)
    }
}
